crates/rspack_core/src/utils/identifier.rs

```diff
@@ -32,17 +32,42 @@ pub fn to_identifier(v: &str) -> Cow<str> {
   }
 }
 
-pub fn stringify_loaders_and_resource<'a>(
+pub async fn stringify_loaders_and_resource<'a>(
   loaders: &'a [ModuleRuleUseLoader],
   resource: &'a str,
+  context: &Context,
+  loader_resolve: &Resolver,
 ) -> Cow<'a, str> {
   if !loaders.is_empty() {
-    let s = loaders
+    let resolve_futures = loaders
       .iter()
-      .map(|i| &*i.loader)
-      .collect::<Vec<_>>()
-      .join("!");
-    Cow::Owned(format!("{s}!{resource}"))
+      .map(|loader| {
+        let loader_request = &loader.loader;
+        async move {
+          (
+            loader_resolve
+              .resolve(context.as_path().as_std_path(), loader_request)
```

Comment:
`resolve` here?

Reply:
yes, should get the absolute path.

Reply:
This is bad for performance, there should have already resolved loader absolute path in somewhere